modules/custom/az_core/az_core.module

```diff
@@ -208,13 +212,41 @@ function az_core_tokens($type, $tokens, array $data, array $options, BubbleableM
         case 'az-canonical-url':
           $replacements[$original] = az_core_canonical_url($data);
           break;
+
+        case 'az-canonical-summary-link-title':
+          $replacements[$original] = az_core_canonical_summary_link_title($data);
+          break;
       }
     }
   }
 
   return $replacements;
 }
 
+/**
+ * Token replacement callback providing the canonical link title.
+ *
+ * Returns the link title value of field_az_link if present on the node, or
+ * "Read more" if not.
+ *
+ * @param array $data
+ *   Node data from hook_tokens().
+ *
+ * @return string
+ *   The node's canononical link title.
+ */
+function az_core_canonical_summary_link_title(array $data) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $data['node'];
+  $read_more_text = 'Read more';
+
+  if (!empty($node->field_az_link[0]->title)) {
+    $read_more_text .= ' at ' . $node->field_az_link[0]->title;
```

Comment:
Hard-coded "Read more" is not ideal. I have one case that caused me to open #1552 where we link directly to a YouTube video and "Read More" doesn't make sense. They instead want it to say something like "View" or "Watch". It'd be better to just replace $read_more_text in this statement with the link title instead of prefixing it with unchangeable text.

Reply:
What if we made the default configurable, and also allowed it to be overridden via the link title field

Reply:
> What if we made the default configurable, and also allowed it to be overridden via the link title field

Actually, yes, I've removed the hard-coded text, thanks